### PR TITLE
TRD correctly shift the trigger bc time to take account of the delay

### PR DESCRIPTION
**Pull request by bazinski**
Correctly shift the trigger bunch crossing according to https://alice.its.cern.ch/jira/browse/CTP-133

### CODE REVIEW

File: Detectors/TRD/reconstruction/src/CruRawReader.cxx [hunk 1]
```diff
@@ -484,6 +484,11 @@ int CruRawReader::processHalfCRU(int cruhbfstartoffset, int numberOfPreviousCRU)
 
   //get eventrecord for event we are looking at
   mIR.bc = mCurrentHalfCRUHeader.BunchCrossing; // correct mIR to have the physics trigger bunchcrossing *NOT* the heartbeat trigger bunch crossing.
+  //shift accordingly
+  mIR.bc = mCurrentHalfCRUHeader.BunchCrossing - o2::ctp::TriggerOffsetsParam::Instance().LM_L0;
+  if (mIR.bc < 0) {
+    mIR.bc = 0;
+  }
```
Comment (shahor02) [inline Detectors/TRD/reconstruction/src/CruRawReader.cxx:491]: Why setting BC to 0 ? The data corresponding to the BC which does not belong to TF should be simply discarded.

Comment (bazinski) [reply Detectors/TRD/reconstruction/src/CruRawReader.cxx:491]: good point, fixing.

Comment (shahor02) [reply Detectors/TRD/reconstruction/src/CruRawReader.cxx:491]: Also, the inverse (adding the bias) should be applied in the mc2raw.

Comment (bazinski) [reply Detectors/TRD/reconstruction/src/CruRawReader.cxx:491]: even better point ;-) I want to get the safety net fixed for the data decoder first, then will come back to this. I dont want to dump the entire dataframe rather only the halfcruheader in question, and with the codes current state its rather delicate.

**Review by shahor02 — APPROVED** [review]
Hi @bazinski 
merging as is since currently the TRD BCs are shifted in the reconstruction, but please add the changes discussed in the comments as additional PR (rejecting trigger data in the BC becomes negative after correction and emulating shift in the mc2raw).

**Pull request merged**